tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc

```diff
@@ -91,10 +92,18 @@ static string GetSmName(std::pair<int, int> compute_capability) {
       {{7, 5}, 75},
   });
   int sm_version = 35;
-  auto it = m->find(compute_capability);
-  if (it != m->end()) {
-    sm_version = it->second;
-  } else {
+  // If the current compute capability isn't known, fallback to the
+  // most recent version before the unknown version.
+  for (auto iter = m->begin(); iter != m->end(); ++iter) {
```

Comment:
Is this just `m->lower_bound(compute_capability)`?

Reply:
lower_bound return an element that is greater or equal then compute_capability.
upper_bound return an element that is greater then compute_capability.

We need an less or equal. So I could use the upper_bound by inverting the sign or changing the comparison function. But I think it would be less readable.
So I didn't change it. If you think it would be more readable or of another function that would do less or equal, tell me.

Reply:
I implemented it and based on the algo suggested in the other commit. It was taking as much line of code and looked less readable to me. So I didn't do it. But it was working.